src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-list/smb-cluster-list.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+
+import _ from 'lodash';
+import { Subscription } from 'rxjs';
+
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { Icons } from '~/app/shared/enum/icons.enum';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableSelection } from '~/app/shared/models/cd-table-selection';
+import { Permission } from '~/app/shared/models/permissions';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { SmbService } from '~/app/shared/api/smb.service';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+
+@Component({
+  selector: 'cd-smb-cluster-list',
+  templateUrl: './smb-cluster-list.component.html',
+  styleUrls: ['./smb-cluster-list.component.scss']
+})
+export class SmbClusterListComponent extends ListWithDetails implements OnInit {
+  @ViewChild('table', { static: true })
+  table: TableComponent;
+
+  columns: CdTableColumn[];
+  permission: Permission;
+  selection = new CdTableSelection();
+  summaryDataSubscription: Subscription;
+  viewCacheStatus: any;
+  smbClusters: any[];
+  tableActions: CdTableAction[];
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    public actionLabels: ActionLabelsI18n,
+    private smbService: SmbService
+  ) {
+    super();
+
+    this.permission = this.authStorageService.getPermissions().smb;
+    this.tableActions = [
+      {
+        name: this.actionLabels.CREATE + ' Cluster ',
+        permission: 'create',
+        icon: Icons.add,
+        click: () => this.openModal(),
+        canBePrimary: (selection: CdTableSelection) => !selection.hasSingleSelection
+      },
+      {
+        name: this.actionLabels.CREATE + ' Share ',
+        permission: 'create',
+        icon: Icons.add,
+        click: () => this.openModal(),
+        canBePrimary: (selection: CdTableSelection) => !selection.hasSelection
+      },
+      {
+        name: this.actionLabels.EDIT,
+        permission: 'update',
+        icon: Icons.edit,
+        click: () => this.openModal()
+      },
+      {
+        name: this.actionLabels.REMOVE,
+        permission: 'read',
+        icon: Icons.bars,
+        click: () => this.openModal()
+      }
+    ];
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`Cluster`,
+        prop: 'cluster_id',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Auth Mode`,
+        prop: 'auth_mode',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Intent`,
+        prop: 'intent',
+        flexGrow: 2
```

Comment:
does this provide anything useful?

Reply:
if its like state, should we map it to the badges or tags?

Reply:
I think we can skip the intent, as it should usually be `present`